sentry_sdk/integrations/django/__init__.py

```diff
@@ -314,6 +316,30 @@ def sentry_patched_asgi_handler(self, receive, send):
     AsgiHandler.__call__ = sentry_patched_asgi_handler
 
 
+def _patch_django_asgi_handler():
+    # type: () -> None
+    try:
+        from django.core.handlers.asgi import ASGIHandler
+    except ImportError:
+        return
+
+    if not HAS_REAL_CONTEXTVARS:
+        # We better have contextvars or we're going to leak state between
+        # requests.
+        #
+        # We cannot hard-raise here because Django may not be used at all in
+        # the current process.
+        logger.warning(
+            "We detected that you are using Django 3. To get proper "
+            "instrumentation for ASGI requests, the Sentry SDK requires "
+            "Python 3.7+ or the aiocontextvars package from PyPI."
+        )
+
+    from sentry_sdk.integrations.django.asgi import patch_django_asgi_handler_impl
```

Comment:
In a wrong combination of Python version and Django, could this potentially blow up?
Maybe wrap this in a `try... except SyntaxError`?

Reply:
I don't think this can blow up because `django.core.handlers.asgi` already would contain invalid syntax (and only exists in Django 3), so the first few lines in this function will return early

fwiw we do test all minor versions of Django in tox.ini, in all Python versions that work with the SDK and Django.